mathics/builtin/combinatorial.py

```diff
@@ -270,3 +270,172 @@ class RogersTanimotoDissimilarity(_BooleanDissimilarity):
+    
+    #> Subsets[{a, b, c}, {3, 4, 5, 6}]
+     : Position 2 of Subsets[{a, b, c}, {3, 4, 5, 6}] must be All, Infinity, a non-negative integer, or a List whose first element (required) is a non-negative integer, second element (optional) is a non-negative integer or Infinity, and third element (optional) is a nonzero integer
+     = Subsets[{a, b, c}, {3, 4, 5, 6}]
+     
+    #> Subsets[{a, b, c}, {-1, 2}]
+     : Position 2 of Subsets[{a, b, c}, {-1, 2}] must be All, Infinity, a non-negative integer, or a List whose first element (required) is a non-negative integer, second element (optional) is a non-negative integer or Infinity, and third element (optional) is a nonzero integer
+     = Subsets[{a, b, c}, {-1, 2}]
+    
+    #> Subsets[{a, b, c}, All]
+     = {{}, {a}, {b}, {c}, {a, b}, {a, c}, {b, c}, {a, b, c}}
+     
+    #> Subsets[{a, b, c}, Infinity]
+     = {{}, {a}, {b}, {c}, {a, b}, {a, c}, {b, c}, {a, b, c}}
+     
+    #> Subsets[{a, b, c}, ALL]
+     : Position 2 of Subsets[{a, b, c}, ALL] must be All, Infinity, a non-negative integer, or a List whose first element (required) is a non-negative integer, second element (optional) is a non-negative integer or Infinity, and third element (optional) is a nonzero integer
+     = Subsets[{a, b, c}, ALL]
+     
+    #> Subsets[{a, b, c}, {a}]
+     : Position 2 of Subsets[{a, b, c}, {a}] must be All, Infinity, a non-negative integer, or a List whose first element (required) is a non-negative integer, second element (optional) is a non-negative integer or Infinity, and third element (optional) is a nonzero integer
+     = Subsets[{a, b, c}, {a}]
+    
+    #> Subsets[{a, b, c}, {}]
+     : Position 2 of Subsets[{a, b, c}, {}] must be All, Infinity, a non-negative integer, or a List whose first element (required) is a non-negative integer, second element (optional) is a non-negative integer or Infinity, and third element (optional) is a nonzero integer
+     = Subsets[{a, b, c}, {}]
+    """
+
+    rules = {
+        'Subsets[list_?ListQ , Pattern[n,_?ListQ|All|DirectedInfinity[1]], spec_]':'Take[Subsets[list, n], spec]'
+        }
+    messages = {
+        'nninfseq': 'Position 2 of `1` must be All, Infinity, a non-negative integer, or a List whose first element (required) is a non-negative integer, second element (optional) is a non-negative integer or Infinity, and third element (optional) is a nonzero integer',
+    }
+    
+    def apply(self, list, evaluation):
+        'Subsets[list_?ListQ]'
+    
+        return self.apply_1(list, Integer(len(list.to_python())), evaluation)
+    
+    def apply_1(self, list, n, evaluation):
+        'Subsets[list_?ListQ, n_]'
+        
+        tlist = [x for x in list.leaves]
+        expr = Expression('Subsets', list, n)
+        
+        if not n.get_head_name() == 'System`Integer' or n.to_python() < 0 :
+            return evaluation.message('Subsets', 'nninfseq', expr)
+        
+        tlen = int(n.to_python())
```

Comment:
we have a utility method for this, `n.get_int_value()` which returns an int or None.